src/Generating/SinglePieceStructuresGen.h

```diff
@@ -0,0 +1,52 @@
+
+// PrefabSingleStructureGen.h
+
+
+/*
+Classes to support the generation of single piece prefab structures
+*/
+
+#pragma once
+
+#include "Globals.h"
+
+#include "ComposableGenerator.h"
+#include "PrefabPiecePool.h"
+
+/** The Single Prefab Structure Generator:
+This uses the cGridStructGen to generate the structures on the map
+This is similar to the Piece Structure Generator but only placing one possible structure
+The Exported cubeset MUST have all possible structures as start structures or the server crashes on generation
```

Comment:
Server crashing on bad (admin) input is a bad idea, can't we fix the code instead?

Reply:
hm...

this would require some changes further down the line. you may ask for the starting piece but if there's no starting piece in the first place you get all pieces. So this would require the load from file function in the piece pool to check if there's a starting piece.

the crash results from the fact that the height generator is a nullpointer in the object responsible for place the piece according to the vertical strategy if there's no starting piece.

so if there is no starting piece we either
a) assign every piece's vertical strategy the generator data
b) refuse to load the cubeset
c) return no pieces and cancel generating

Reply:
this error shouldn't occure only in the new structures but also in the others if there is no starting structure defined so it would require some changes in the prefal pool class

Reply:
i opened a issue for that. If you think this is related enough i can fix this in this PR

Reply:
Nah, a separate PR it is. But since you already investigated this, it'd be great if you fixed it too, as long as the memory's still fresh.